Generator/generator-botbuilder-java/generators/app/index.js

```diff
@@ -1,49 +1,268 @@
+                this.templatePath(path.join(template, 'tree', '**')),
+                this.destinationPath(),
+                {
+                    botName,
+                    packageName
+                }
+            );
+
+            // Copy main source
+            this.fs.copyTpl(
+                this.templatePath(path.join(template, 'src/main/java/**')),
+                this.destinationPath(path.join('src/main/java', packageTree)),
+                {
+                    packageName
+                }
+            );
+
+            // Copy test source
+            this.fs.copyTpl(
+                this.templatePath(path.join(template, 'src/test/java/**')),
+                this.destinationPath(path.join('src/test/java', packageTree)),
+                {
+                    packageName
+                }
+            );
+        }
+    }
+
+    end() {
+        if(this.templateConfig.finalConfirmation === true) {
+            this.log(chalk.green('------------------------ '));
+            this.log(chalk.green(' Your new bot is ready!  '));
+            this.log(chalk.green('------------------------ '));
+            this.log(`Your bot should be in a directory named ${_.kebabCase(this.templateConfig.botName)}`);
+            this.log('Open the ' + chalk.green.bold('README.md') + ' to learn how to run your bot. ');
+            this.log('Thank you for using the Microsoft Bot Framework. ');
+            this.log('\n< ** > The Bot Framework Team');
+        } else {
+            this.log(chalk.red.bold('-------------------------------- '));
+            this.log(chalk.red.bold(' New bot creation was canceled. '));
+            this.log(chalk.red.bold('-------------------------------- '));
+            this.log('Thank you for using the Microsoft Bot Framework. ');
+            this.log('\n< ** > The Bot Framework Team');
+        }
+    }
+
+    _configureCommandlineOptions() {
+        this.option('botName', {
+            desc: 'The name you want to give to your bot',
+            type: String,
+            default: 'echo',
+            alias: 'N'
+        });
+
+        this.option('packageName', {
+            desc: `What's the fully qualified package name of your bot?`,
+            type: String,
+            default: 'com.mycompany.echo',
+            alias: 'P'
+        });
+
+        const templateDesc = `The initial bot capabilities. (${BOT_TEMPLATE_NAME_EMPTY} | ${BOT_TEMPLATE_NAME_SIMPLE} | ${BOT_TEMPLATE_NAME_CORE})`;
+        this.option('template', {
+            desc: templateDesc,
+            type: String,
+            default: BOT_TEMPLATE_NAME_SIMPLE,
+            alias: 'T'
+        });
+
+        this.argument('noprompt', {
+            desc: 'Do not prompt for any information or confirmation',
+            type: Boolean,
+            required: false,
+            default: false
+        });
+    }
+
+    _getPrompts() {
+        const noprompt = this.options.noprompt;
+        const prompts = {
```

Comment:
I suggested rewriting this method to return an array. Something like:

```js
_getPrompts() {
  const noprompt = this.options.noprompt;
  return [
    async () => {
      if (noprompt) {
        return;
      }
      ...
    },
    async () => { ... },
  ];
}
```

Marking a function `async` implicitly wraps it in a Promise so you never have to do `return Promise.resolve();`, simply `return`.